snapcraft/internal/repo/snaps.py

```diff
@@ -0,0 +1,210 @@
+        self.name, self.channel = _get_parsed_snap(snap)
+        self._original_channel = self.channel
+        if not self.channel or self.channel == 'stable':
+            self.channel = 'latest/stable'
+
+        # This store information from a local request
+        self._local_snap_info = None
+        # And this stores information from a remote request
+        self._store_snap_info = None
+
+        self._is_installed = None
+        self._is_on_store = None
+
+    @property
+    def installed(self):
+        if self._is_installed is None:
+            self._is_installed = self.get_local_snap_info() is not None
+        return self._is_installed
+
+    @property
+    def on_store(self):
+        if self._is_on_store is None:
+            self._is_on_store = self.get_store_snap_info() is not None
+        return self._is_on_store
+
+    def get_local_snap_info(self):
+        """Returns a local payload for the snap.
+
+        Validity of the results are determined by checking self.installed."""
+        if self._is_installed is None:
+            with contextlib.suppress(exceptions.HTTPError):
+                self._local_snap_info = _get_local_snap_info(self.name)
+        return self._local_snap_info
+
+    def get_store_snap_info(self):
+        """Returns a store payload for the snap.
+
+        Validity of the results are determined by checking self.remote."""
+        if self._is_on_store is None:
+            with contextlib.suppress(exceptions.HTTPError):
+                self._store_snap_info = _get_store_snap_info(self.name)
+        return self._store_snap_info
+
+    def _get_store_channels(self):
+        snap_store_info = self.get_store_snap_info()
+        if not self.on_store:
+            return dict()
+
+        return snap_store_info['channels']
+
+    def get_current_channel(self):
+        current_channel = ''
+        if self.installed:
+            local_snap_info = self.get_local_snap_info()
+            current_channel = local_snap_info['channel']
+            if current_channel == 'stable':
+                current_channel = 'latest/stable'
+        return current_channel
+
+    def is_classic(self):
+        store_channels = self._get_store_channels()
+        return store_channels[self.channel]['confinement'] == 'classic'
+
+    def is_valid(self):
+        """Check if the snap is valid."""
+        if not self.on_store:
+            return False
+        store_channels = self._get_store_channels()
+        return self.channel in store_channels.keys()
+
+    def install(self):
+        """Installs the snap onto the system."""
+        snap_install_cmd = []
+        if _snap_command_requires_sudo():
+            snap_install_cmd = ['sudo']
+        snap_install_cmd.extend(['snap', 'install', self.name])
+        if self._original_channel:
+            snap_install_cmd.extend(['--channel', self._original_channel])
+        if self.is_classic:
```

Comment:
this is missing the TODO comment to make it a user choice on non ci/container environments

Reply:
this is done in the `is_classic` check now